src/server/service/kv.rs

```diff
@@ -862,6 +913,10 @@ impl<E: Engine, L: LockManager, F: KvFormat> Tikv for Service<E, L, F> {
         stream: RequestStream<SnapshotChunk>,
         sink: ClientStreamingSink<Done>,
     ) {
+        if let Some(true) = self.raft_message_filter.should_reject_snapshot() {
+            RAFT_SNAPSHOT_REJECTS.inc();
```

Comment:
Why don't call `sink.fail`?

Reply:
Which status should I use?

Reply:
Meanwhile, I found rejection for masAppend wil just return `Ok(())`, so I do not understand why we need to call `sink.fail` here if we want to reject the snapshot and following msgSnapshot.

Reply:
RpcStatus::with_message(RpcStatusCode::RESOURCE_EXHAUSTED, "msg")

Reply:
> Meanwhile, I found rejection for msgAppend wil just return Ok(()), so I do not understand why we need to call sink.fail here if we want to reject the snapshot and following msgSnapshot. 

We can drop raft message silently because raft protocol can help us handle the necessary message resend. But `snapshot` is not raft, so we need to fail the stream to tell the remote the snapshot data sent failed.

Reply:
So when will the snapshot being resend?

Reply:
When you report rejected

Reply:
Got it, thx~